CMakeLists.txt

```diff
@@ -39,6 +39,19 @@ if(GIT_FOUND)
      OUTPUT_VARIABLE GIT_COMMIT_SHORT
      OUTPUT_STRIP_TRAILING_WHITESPACE
    )
+
+   execute_process(
+     COMMAND ${GIT_EXECUTABLE} describe --tags
+     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+     OUTPUT_VARIABLE GIT_TAG
+     OUTPUT_STRIP_TRAILING_WHITESPACE
+   )
+
+   string( REGEX REPLACE "^([^.]+)\\.([^.]+)\\.(.+)$" "\\1" RES_VERSION_MAJOR "${GIT_TAG}")
+   string( REGEX REPLACE "^([^.]+)\\.([^.]+)\\.(.+)$" "\\2" RES_VERSION_MINOR "${GIT_TAG}")
+   string( REGEX REPLACE "^([^.]+)\\.([^.]+)\\.(.+)$" "\\3" RES_VERSION_MICRO "${GIT_TAG}")
+
+   message( STATUS "libres version: ${GIT_TAG}" )
```

Comment:
I think it is also a good time to get rid of the hard-coded version above, and set to N/A or similar if git is not available.